ProjFS.Mac/PrjFSLib/PrjFSLib.cpp

```diff
@@ -1122,10 +1123,10 @@ static PrjFS_Result HandleFileNotification(
         notificationType,
         nullptr /* destinationRelativePath */);
     
-    // We have to convert to full for PreDelete until we have WILL_RENAME support.  The PreDelete could be for a rename.
+    // We have to convert to full for PrjFS_NotificationType_PreDeleteFromRename in order to move the file
```

Comment:
Minor:  This comment isn't quite accurate, as the file could be moved even if we don't convert it to full.

What if we went with something like:

```
// Remove placeholder xattrs for renames (PreDeleteFromRename) because:
//   - Renames are treated as "delete old path" + "create new path", and new files should not be marked as placeholders
//   - The target of the rename might be outside of the root, and we should not allow placeholder files to 
//     live outside of virtualization roots (as the kext won't know what provider they below to)